src/native/corehost/deps_format.cpp

```diff
@@ -439,7 +439,7 @@ bool deps_json_t::has_package(const pal::string_t& name, const pal::string_t& ve
 bool deps_json_t::load(bool is_framework_dependent, const pal::string_t& deps_path, const rid_fallback_graph_t& rid_fallback_graph)
 {
     m_deps_file = deps_path;
-    m_file_exists = bundle::info_t::config_t::probe(deps_path) || pal::file_exists(deps_path);
+    m_file_exists = !m_deps_file.empty() && (pal::realpath(&m_deps_file, true) || bundle::info_t::config_t::probe(deps_path));
```

Comment:
Not sure how much it matters in practice, but we used to do the bundle probe first (which would bail if it wasn't single-file), but now we're always doing `realpath` first, so we always pay that cost in single-file?

Reply:
I reverted the order so we don't pay extra when running on single-file.